artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/Page.java

```diff
@@ -119,12 +117,6 @@ public void setLiveCache(LivePageCache pageCache) {
       return messages;
    }
 
-   private static void decodeInto(ByteBuffer fileBuffer, int encodedSize, PagedMessageImpl msg) {
-      final ActiveMQBuffer wrappedBuffer = ActiveMQBuffers.wrappedBuffer(fileBuffer);
```

Comment:
This was wrapping each message basically?

I see.. that's a nice improvement

Reply:
Yep! Now I just need to fix something on the wrapper indexes and it should be fine...I didn't expect to allocate so much garbage just for few instances :O

Reply:
Ok, I've just fixed the missing bits (due to how `ByteBuf` work)